pyproject.toml

```diff
@@ -167,6 +167,27 @@ extend_skip_glob = ["asdf/extern/*", ".eggs/*", ".tox/*"]
 
 [tool.ruff]
 line-length = 120
+select = ["ALL"]
+extend-ignore = [
+    # Ignored check groups
+    "C90", # mccabe complexity
+    "D", # pydocstyle
+    "ANN", # flake8-annotations
+    "FBT", # flake8-boolean-trap
+    "EM", # flake8-errmsg
```

Comment:
We should definitely fix the error messages for these checks because they are designed to help with having slightly better error tracebacks . Such as, python tends to return the line the error was raised on, so if the message is in that line then we duplicate information presented to the user which is confusing.

Reply:
The errors found by this are:
```shell

asdf/_display.py:41:31: FBT002 Boolean default value in function definition
asdf/_display.py:102:51: FBT003 Boolean positional value in function call
asdf/_display.py:222:71: FBT003 Boolean positional value in function call
asdf/_node_info.py:23:87: FBT002 Boolean default value in function definition
asdf/_node_info.py:55:69: FBT002 Boolean default value in function definition
asdf/_node_info.py:55:101: FBT002 Boolean default value in function definition
asdf/_node_info.py:175:72: FBT002 Boolean default value in function definition
asdf/_node_info.py:175:87: FBT002 Boolean default value in function definition
asdf/_node_info.py:240:101: FBT002 Boolean default value in function definition
asdf/_node_info.py:295:42: FBT002 Boolean default value in function definition
asdf/asdf.py:58:33: FBT002 Boolean default value in function definition
asdf/asdf.py:59:33: FBT002 Boolean default value in function definition
asdf/asdf.py:60:36: FBT002 Boolean default value in function definition
asdf/asdf.py:61:21: FBT002 Boolean default value in function definition
asdf/asdf.py:62:19: FBT002 Boolean default value in function definition
asdf/asdf.py:135:92: FBT003 Boolean positional value in function call
asdf/asdf.py:135:98: FBT003 Boolean positional value in function call
asdf/asdf.py:283:46: FBT002 Boolean default value in function definition
asdf/asdf.py:610:38: FBT002 Boolean default value in function definition
asdf/asdf.py:610:52: FBT002 Boolean default value in function definition
asdf/asdf.py:794:28: FBT002 Boolean default value in function definition
asdf/asdf.py:796:27: FBT002 Boolean default value in function definition
asdf/asdf.py:797:26: FBT002 Boolean default value in function definition
asdf/asdf.py:798:32: FBT002 Boolean default value in function definition
asdf/asdf.py:799:35: FBT002 Boolean default value in function definition
asdf/asdf.py:894:28: FBT002 Boolean default value in function definition
asdf/asdf.py:896:27: FBT002 Boolean default value in function definition
asdf/asdf.py:897:26: FBT002 Boolean default value in function definition
asdf/asdf.py:898:32: FBT002 Boolean default value in function definition
asdf/asdf.py:899:35: FBT002 Boolean default value in function definition
asdf/asdf.py:929:28: FBT002 Boolean default value in function definition
asdf/asdf.py:931:27: FBT002 Boolean default value in function definition
asdf/asdf.py:932:26: FBT002 Boolean default value in function definition
asdf/asdf.py:933:32: FBT002 Boolean default value in function definition
asdf/asdf.py:934:35: FBT002 Boolean default value in function definition
asdf/asdf.py:989:28: FBT002 Boolean default value in function definition
asdf/asdf.py:991:33: FBT002 Boolean default value in function definition
asdf/asdf.py:992:33: FBT002 Boolean default value in function definition
asdf/asdf.py:993:26: FBT002 Boolean default value in function definition
asdf/asdf.py:994:21: FBT002 Boolean default value in function definition
asdf/asdf.py:995:19: FBT002 Boolean default value in function definition
asdf/asdf.py:997:32: FBT002 Boolean default value in function definition
asdf/asdf.py:998:35: FBT002 Boolean default value in function definition
asdf/asdf.py:1100:42: FBT003 Boolean positional value in function call
asdf/asdf.py:1115:20: FBT002 Boolean default value in function definition
asdf/asdf.py:1116:29: FBT002 Boolean default value in function definition
asdf/asdf.py:1260:20: FBT002 Boolean default value in function definition
asdf/asdf.py:1261:29: FBT002 Boolean default value in function definition
asdf/asdf.py:1393:53: FBT002 Boolean default value in function definition
asdf/asdf.py:1542:71: FBT002 Boolean default value in function definition
asdf/asdf.py:1542:103: FBT002 Boolean default value in function definition
asdf/asdf.py:1577:35: FBT002 Boolean default value in function definition
asdf/asdf.py:1719:24: FBT002 Boolean default value in function definition
asdf/asdf.py:1721:29: FBT002 Boolean default value in function definition
asdf/asdf.py:1722:29: FBT002 Boolean default value in function definition
asdf/asdf.py:1723:22: FBT002 Boolean default value in function definition
asdf/asdf.py:1724:17: FBT002 Boolean default value in function definition
asdf/asdf.py:1725:15: FBT002 Boolean default value in function definition
asdf/asdf.py:1727:28: FBT002 Boolean default value in function definition
asdf/asdf.py:1728:31: FBT002 Boolean default value in function definition
asdf/asdf.py:1729:13: FBT002 Boolean default value in function definition
asdf/block.py:24:46: FBT002 Boolean default value in function definition
asdf/block.py:24:63: FBT002 Boolean default value in function definition
asdf/block.py:216:56: FBT002 Boolean default value in function definition
asdf/block.py:225:51: FBT002 Boolean default value in function definition
asdf/block.py:225:77: FBT002 Boolean default value in function definition
asdf/block.py:278:77: FBT003 Boolean positional value in function call
asdf/block.py:284:59: FBT002 Boolean default value in function definition
asdf/block.py:340:53: FBT002 Boolean default value in function definition
asdf/block.py:644:81: FBT003 Boolean positional value in function call
asdf/block.py:798:78: FBT002 Boolean default value in function definition
asdf/block.py:798:94: FBT002 Boolean default value in function definition
asdf/block.py:972:35: FBT002 Boolean default value in function definition
asdf/block.py:972:60: FBT002 Boolean default value in function definition
asdf/block.py:1238:43: FBT002 Boolean default value in function definition
asdf/block.py:1238:59: FBT002 Boolean default value in function definition
asdf/commands/defragment.py:49:56: FBT002 Boolean default value in function definition
asdf/commands/diff.py:149:56: FBT002 Boolean default value in function definition
asdf/commands/diff.py:183:65: FBT002 Boolean default value in function definition
asdf/commands/diff.py:183:86: FBT002 Boolean default value in function definition
asdf/commands/diff.py:183:104: FBT002 Boolean default value in function definition
asdf/commands/diff.py:226:45: FBT003 Boolean positional value in function call
asdf/commands/diff.py:227:45: FBT003 Boolean positional value in function call
asdf/commands/diff.py:290:67: FBT003 Boolean positional value in function call
asdf/commands/diff.py:292:67: FBT003 Boolean positional value in function call
asdf/commands/edit.py:127:67: FBT003 Boolean positional value in function call
asdf/commands/exploded.py:54:53: FBT002 Boolean default value in function definition
asdf/commands/tags.py:47:31: FBT002 Boolean default value in function definition
asdf/commands/tests/test_diff.py:14:57: FBT002 Boolean default value in function definition
asdf/commands/tests/test_diff.py:68:64: FBT003 Boolean positional value in function call
asdf/commands/tests/test_extension.py:14:21: FBT003 Boolean positional value in function call
asdf/commands/tests/test_extension.py:14:27: FBT003 Boolean positional value in function call
asdf/commands/to_yaml.py:52:53: FBT002 Boolean default value in function definition
asdf/compression.py:168:67: FBT002 Boolean default value in function definition
asdf/fits_embed.py:167:28: FBT002 Boolean default value in function definition
asdf/fits_embed.py:169:33: FBT002 Boolean default value in function definition
asdf/fits_embed.py:170:33: FBT002 Boolean default value in function definition
asdf/fits_embed.py:171:32: FBT002 Boolean default value in function definition
asdf/fits_embed.py:172:35: FBT002 Boolean default value in function definition
asdf/fits_embed.py:236:28: FBT002 Boolean default value in function definition
asdf/fits_embed.py:238:33: FBT002 Boolean default value in function definition
asdf/fits_embed.py:239:33: FBT002 Boolean default value in function definition
asdf/fits_embed.py:240:32: FBT002 Boolean default value in function definition
asdf/fits_embed.py:241:35: FBT002 Boolean default value in function definition
asdf/fits_embed.py:300:78: FBT002 Boolean default value in function definition
asdf/fits_embed.py:300:99: FBT002 Boolean default value in function definition
asdf/fits_embed.py:324:20: FBT002 Boolean default value in function definition
asdf/fits_embed.py:325:23: FBT002 Boolean default value in function definition
asdf/fits_embed.py:341:85: FBT002 Boolean default value in function definition
asdf/generic_io.py:107:76: FBT002 Boolean default value in function definition
asdf/generic_io.py:107:114: FBT002 Boolean default value in function definition
asdf/generic_io.py:182:40: FBT002 Boolean default value in function definition
asdf/generic_io.py:421:72: FBT002 Boolean default value in function definition
asdf/generic_io.py:421:109: FBT002 Boolean default value in function definition
asdf/generic_io.py:480:72: FBT002 Boolean default value in function definition
asdf/generic_io.py:480:109: FBT002 Boolean default value in function definition
asdf/generic_io.py:519:72: FBT002 Boolean default value in function definition
asdf/generic_io.py:519:109: FBT002 Boolean default value in function definition
asdf/generic_io.py:671:72: FBT002 Boolean default value in function definition
asdf/generic_io.py:671:109: FBT002 Boolean default value in function definition
asdf/generic_io.py:729:40: FBT002 Boolean default value in function definition
asdf/generic_io.py:801:44: FBT002 Boolean default value in function definition
asdf/generic_io.py:841:72: FBT002 Boolean default value in function definition
asdf/generic_io.py:841:109: FBT002 Boolean default value in function definition
asdf/generic_io.py:879:34: FBT002 Boolean default value in function definition
asdf/generic_io.py:952:46: FBT002 Boolean default value in function definition
asdf/schema.py:247:70: FBT002 Boolean default value in function definition
asdf/schema.py:248:100: FBT003 Boolean positional value in function call
asdf/schema.py:248:107: FBT003 Boolean positional value in function call
asdf/schema.py:411:56: FBT002 Boolean default value in function definition
asdf/schema.py:411:82: FBT002 Boolean default value in function definition
asdf/schema.py:507:68: FBT003 Boolean positional value in function call
asdf/schema.py:636:72: FBT002 Boolean default value in function definition
asdf/schema.py:684:42: FBT002 Boolean default value in function definition
asdf/schema.py:718:43: FBT002 Boolean default value in function definition
asdf/schema.py:760:89: FBT003 Boolean positional value in function call
asdf/schema.py:760:96: FBT003 Boolean positional value in function call
asdf/search.py:317:60: FBT002 Boolean default value in function definition
asdf/search.py:317:92: FBT002 Boolean default value in function definition
asdf/tags/core/ndarray.py:94:59: FBT002 Boolean default value in function definition
asdf/tests/helpers.py:285:45: FBT002 Boolean default value in function definition
asdf/tests/test_reference_files.py:33:54: FBT002 Boolean default value in function definition
asdf/tests/test_schema.py:1076:45: FBT003 Boolean positional value in function call
asdf/tests/test_schema.py:1080:45: FBT003 Boolean positional value in function call
asdf/treeutil.py:221:63: FBT002 Boolean default value in function definition
asdf/treeutil.py:221:80: FBT002 Boolean default value in function definition
asdf/types.py:204:44: FBT002 Boolean default value in function definition
asdf/util.py:319:34: FBT002 Boolean default value in function definition
asdf/util.py:337:43: FBT002 Boolean default value in function definition
asdf/yamlutil.py:272:59: FBT002 Boolean default value in function definition
compatibility_tests/test_file_compatibility.py:34:42: FBT002 Boolean default value in function definition
pytest_asdf/plugin.py:129:64: FBT002 Boolean default value in function definition
pytest_asdf/plugin.py:200:33: FBT002 Boolean default value in function definition
pytest_asdf/plugin.py:201:33: FBT002 Boolean default value in function definition
Found 154 error(s).
```